server/client/src/Components/RegisterForm/RegisterForm.component.jsx

```diff
@@ -0,0 +1,168 @@
+import React , {useState, useEffect}from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { useHttp } from '../../Hooks/http.hook';
+
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import Snackbar from '@mui/material/Snackbar';
+import MuiAlert from '@mui/material/Alert';
+
+import './RegisterForm.style.css';
+
+function RegisterForm() {
+    const {loading, error, request, clearError}= useHttp();
```

Comment:
Replace all logic to the AuthPage this component should only take care of presentation